MetadataExtractor/Formats/Png/PngMetadataReader.cs

```diff
@@ -84,10 +84,26 @@ public static DirectoryList ReadMetadata([NotNull] string filePath)
         [NotNull]
         public static DirectoryList ReadMetadata([NotNull] Stream stream)
         {
-            return new PngChunkReader()
-                .Extract(new SequentialStreamReader(stream), _desiredChunkTypes)
-                .SelectMany(ProcessChunk)
-                .ToList();
+            List<Directory> directories = null;
+
+            var chunks = new PngChunkReader().Extract(new SequentialStreamReader(stream), _desiredChunkTypes);
+
+            foreach (var chunk in chunks)
+            {
+                if(directories == null)
+                    directories = new List<Directory>();
```

Comment:
As an aside, I'd like to upgrade to C# 8 soon, after which we'll be able to write `directories ??= new List<Directory>()`.